packages/mui-utils/src/resolveProps/resolveProps.ts

```diff
@@ -14,36 +16,51 @@ export default function resolveProps<
 >(defaultProps: T, props: T) {
   const output = { ...props };
 
-  (Object.keys(defaultProps) as Array<keyof T>).forEach((propName) => {
-    if (propName.toString().match(/^(components|slots)$/)) {
-      output[propName] = {
-        ...(defaultProps[propName] as any),
-        ...(output[propName] as any),
-      };
-    } else if (propName.toString().match(/^(componentsProps|slotProps)$/)) {
-      const defaultSlotProps = (defaultProps[propName] || {}) as T[keyof T];
-      const slotProps = props[propName] as {} as T[keyof T];
-      output[propName] = {} as T[keyof T];
```

Comment:
This property is redefined in all 3 branches below, which are exhaustive, so the empty object allocated here is never used.